kuma/javascript/src/payments/feedback-form.jsx

```diff
@@ -0,0 +1,106 @@
+import { gettext, Interpolated } from '../l10n.js';
+import { getCookie } from '../utils.js';
+
+export const FEEDBACK_URL = '/payments/feedback';
+export const MIN_STRING_LENGTH = 10;
+
+const FeedbackForm = (): React.Node => {
+    const [feedback, setFeedback] = React.useState<string>('');
+    const [status, setStatus] = React.useState<'success' | 'idle'>('idle');
+    const [error, setError] = React.useState<React.Node | null>(null);
+
+    const handleChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+        setFeedback(value);
+    };
+
+    const handleSubmit = (event: SyntheticEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        const trimmedFeedback = feedback.trim();
+
+        // Feedback should be greater than 10 characters
+        if (trimmedFeedback.length < MIN_STRING_LENGTH) {
+            setError(
+                `To ensure more constructive feedback, a minimum of ${MIN_STRING_LENGTH} characters is required.`
+            );
+            return false;
+        }
+
+        fetch(FEEDBACK_URL, {
+            method: 'POST',
+            body: JSON.stringify({ feedback: trimmedFeedback }),
+            headers: {
+                'Content-Type': 'application/json',
+                'X-CSRFToken': getCookie('csrftoken')
+            }
+        })
+            .then(res => {
+                // Remove focus from button or input
+                if (document.activeElement) {
+                    document.activeElement.blur();
+                }
+
+                if (!res.ok) {
+                    throw new Error(
+                        'Request (POST) to /api/v1/payments/feedback failed'
+                    );
+                }
+                return res;
+            })
+            .then(() => {
+                // Clear form, show thank you message
+                setFeedback('');
+                setError('');
+                setStatus('success');
+            })
+            .catch(() => {
+                setError(
+                    <Interpolated
+                        id={gettext(
+                            "We're sorry, something went wrong. Please try again or send your feedback to <emailLink />."
+                        )}
+                        emailLink={
+                            <a
+                                target="_blank"
+                                rel="noopener noreferrer"
+                                href="mailto:mdn-support@mozilla.com"
+                            >
+                                {gettext('mdn-support@mozilla.com')}
+                            </a>
+                        }
+                    />
+                );
+            });
+    };
+
+    return (
+        <form onSubmit={handleSubmit}>
+            <input
+                data-testid="feedback-input"
+                type="text"
+                placeholder={gettext('Enter optional feedback…')}
```

Comment:
This still doesn't make sense. Or am I missing something? The `<input required>` means the browser will attempt to block the `onsubmit` if the `input` is left empty.

Reply:
It's optional to submit feedback, so the user can choose ignore the form if they want to. I think it makes more sense in the context of the whole page: 

<img width="1121" alt="Screen Shot 2020-03-30 at 10 19 46 PM" src="https://user-images.githubusercontent.com/650/77980154-97955180-72d4-11ea-87ce-83940e51545b.png">

Does that help?